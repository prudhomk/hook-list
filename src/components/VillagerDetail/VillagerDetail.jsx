/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';

const VillagerDetail = ({ name, species, phrase, quote, personality, birthday, skill, image }) => {

  return (
    <>
      <p>{ name }</p>
      <img src={ image }></img>
      <p>{ species }</p>
      <p>{ phrase }</p>
      <p>{ quote }</p>
      <p>{ personality }</p>
      <p>{ birthday }</p>
      <p>{ skill }</p>
    </>
  );
};

VillagerDetail.propTypes = {
  _id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  species: PropTypes.string.isRequired,
  phrase: PropTypes.string.isRequired,
  quote: PropTypes.string.isRequired,
  personality: PropTypes.string.isRequired,
  birthday: PropTypes.string.isRequired,
  skill: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};

export default VillagerDetail;
