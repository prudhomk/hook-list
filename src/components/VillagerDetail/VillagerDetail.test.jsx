/* eslint-disable max-len */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import VillagerDetail from './VillagerDetail';
import { MemoryRouter } from 'react-router-dom';

describe('Villager Detail component', () => {
  afterEach(() => cleanup());
  it('renders Villager Details', () => {
    const { asFragment } = render(<MemoryRouter><VillagerDetail /></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});
