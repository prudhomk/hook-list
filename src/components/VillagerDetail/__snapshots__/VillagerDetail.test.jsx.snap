// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Villager Detail component renders Villager Details 1`] = `
<DocumentFragment>
  <p />
  <img />
  <p />
  <p />
  <p />
  <p />
  <p />
  <p />
</DocumentFragment>
`;
